### ISSUE TITLE
The Elder Scrolls IV: Oblivion cannot use Anti-Aliasing or Anisotropic Filtering outside of dxvk via Nvidia Control Panel, or within the dxvk conf file. 

### ISSUE THREAD

**Issue by Darius851** [doitsujin/dxvk#3786]
It appears that the The Elder Scrolls IV: Oblivion cannot use Anti-Aliasing or Anisotropic Filtering outside of dxvk via Nvidia Control Panel, or within the dxvk conf file. 

Can anyone please help? Also, is dxvk compatible with Oblivion Reloaded 6.5?

### System information
- O.S Windows 11 x64
- GPU: RTX 4090
- Driver: 537.58
- Wine version: N/A
- DXVK version: v2.3 - x32 - d3d9 dll

Edit: It definitely doesn't work, turning off Oblivion Reloaded's Anisotropic Settings and setting them through the DXVK Conf File does nothing.

**Comment by Blisto91** [comment]
> via Nvidia Control Panel

That is expected as i'm pretty sure those settings only work for older APIs like d3d9.

> It definitely doesn't work, turning off Oblivion Reloaded's Anisotropic Settings and setting them through the DXVK Conf File does nothing.

Make sure the setting is picked up by dxvk by looking at the logs. The setting is working fine here

<details>
  <summary>Screenshots</summary>
  
Compare full image for more detail

Default game
![withoutAF](https://github.com/doitsujin/dxvk/assets/47954800/bec23ed1-c6cf-4a8c-bc06-8467bb92efed)

`d3d9.samplerAnisotropy = 16`
![withAF](https://github.com/doitsujin/dxvk/assets/47954800/eb3e6d40-49fe-46c9-8675-084d31f3100c)
</details>

As for the msaa setting it also works for me. Note that you have to turn on AA in the game itself also or it will break the game rendering. E.g setting it to 2 and then to 8 via dxvk.
Note that it breaking the game is not considered a bug as the configs are mostly meant for debugging and not attempting to do any advanced magic.

**Comment by Darius851** [comment]
I've been trying, but it seems I have had no luck enabling the
d3d9.samplerAnisotropy setting.

Nvidia Control Panel is at its defaults, Oblivion Reloaded's anisotropic
settings have been turned off, but the setting for DXVK persist's in
staying off.

"As for the msaa setting it also works for me. Note that you have to turn
on AA in the game itself also or it will break the game rendering."

That is something I was hoping to avoid, turning on the AA settings in-game
as I had desired to maintain AA, with HDR both enabled. Hence the setting
in the NV Utility.

Is there any way for me to see the DXVK settings in-game though, like what
you have?

**Comment by Blisto91** [comment]
> Is there any way for me to see the DXVK settings in-game though, like what
you have?

You have to check the dxvk log which should appear next to the exe. It will tell if have configs have been found and applied.
Should show something like this
```
info:  Game: Oblivion.exe
info:  DXVK: v2.3-26-g5e06cf95737ccf8
info:  Found config file: dxvk.conf
info:  Effective configuration:
info:    d3d9.samplerAnisotropy = 16
```

If you mean the hud i have on my screen then that doesn't tell about the configs. Only various debug information and fps.

**Comment by Darius851** [comment]
So, I checked the d3d9 log. The game itself is detected, but the settings made to DXVK unfortunately are not.

Now, to my embarrassment. I had the  dxvk file named as dxvk.conf.txt, I'm assuming this would make the file unrecognizable by the DXVK plugin, I have since remediated this. But, it appears the changes are still not applying.

Instead of getting your values
info:    d3d9.samplerAnisotropy = 16

I'm getting this, 
info:    samplerAnisotropy                      : 1

Any idea as to why my line looks so different?

**Comment by Blisto91** [comment]
Send your config

**Comment by Darius851** [comment]
Here's what I got.
[Oblivion_d3d9.log](https://github.com/doitsujin/dxvk/files/13856897/Oblivion_d3d9.log)

[Setting Sample.txt](https://github.com/doitsujin/dxvk/files/13856904/Setting.Sample.txt)

**Comment by Blisto91** [comment]
If you have the `#` included in your config then delete it as that will make the line a comment that is ignored.
The line should just be e.g `d3d9.samplerAnisotropy = 8`

**Comment by Darius851** [comment]
Uncommenting did work, thank you!

One more question, if I wanted to have Oblivion's equivalent to 4x Anti-Aliasing, what should I set MSAA to in DXVK?

**Comment by K0bin** [comment]
> One more question, if I wanted to have Oblivion's equivalent to 4x Anti-Aliasing, what should I set MSAA to in DXVK?

Nothing. There is no option in DXVK to force MSAA in Oblivion. The option that exists only works for very simple games.

**Comment by billbeans** [comment]
Hey sorry if this is a dumb question but are classic vanilla Oblivion landscapes supposed to look this bad or is my AF setting not working properly?

<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/c9259e24-51ff-41af-828e-78e9c091f556" />

It appears that the setting is being loaded properly by DXVK as I get the following in my log:
```
info:  Game: Oblivion.exe
info:  DXVK: v2.7.1-509-g1676dcaf342a9b1
info:  Build: x86 gcc 14.2.0
info:  Found config file: dxvk.conf
info:  Effective configuration:
info:    d3d9.samplerAnisotropy = 16
```
Here's the full log file: [Oblivion_d3d9.log](https://github.com/user-attachments/files/29249627/Oblivion_d3d9.log)

Some posts I found online were saying that the draw distance should be better if you enable anisotropic filtering for your GPU, but I think it looks the exact same as before I created the dxvk.conf file. Is there something else I'm supposed to do? I don't remember it looking that bad, even for such an old game.

**Comment by K0bin** [comment]
That has nothing to do with anisotropic filtering. The game is simply using very low resolution textures for distant terrain.

Remember that the game is 20 years old and designed for the hardware of that time.
Looking at videos and screenshots of the game, this simply appears to be how the game works.

**Comment by billbeans** [comment]
Yeah, I think you're right. I just can't believe it would step from high res to ultra low res like that, it looks terrible. It's also weird how the trees look so detailed yet they're placed on top of blurry terrain, this is the main thing that seems 'off' to me. I played the game only a few years ago and I don't remember it being so noticeable.

I've learned in the past couple of hours that there is a `uGridsToLoad` setting in the Oblivion.ini file that makes the game render terrain farther, but it is supposedly very unstable and not recommended to change. I bumped it from the default 5 to 11 and sure enough, it renders a bit farther, but there is some glitching in the water reflections. I think I will leave it back at 5 for now.

<details>
  <summary>Screenshot</summary>
  
<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/c80967a3-81b7-46f6-b7dc-8be3fb49327f" />

</details>

I should also mention that I'm playing from an original retail CD. Maybe the GOG GOTY edition has better draw distance. Either way, this seems to have nothing to do with dxvk.
